pkg/syncer/spec/spec_controller.go

```diff
@@ -96,7 +99,56 @@ func NewSpecSyncer(gvrs []schema.GroupVersionResource, upstreamClusterName logic
 				c.AddToQueue(gvr, obj)
 			},
 		})
-		klog.InfoS("Set up informer", "clusterName", upstreamClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
+		klog.InfoS("Set up upstream informer", "clusterName", upstreamClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
+
+		downstreamInformers.ForResource(gvr).Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+			DeleteFunc: func(obj interface{}) {
+				unstrob := obj.(*unstructured.Unstructured)
+				klog.InfoS("downstream delete event", "gvr", gvr, "namespace", unstrob.GetNamespace(), "name", unstrob.GetName())
+				//if there is no label then it is not owned by kcp
+				if unstrob.GetLabels() == nil {
+					return
+				}
+				//Use namespace lister
+				namespaces := downstreamClient.Resource(schema.GroupVersionResource{
+					Group:    "",
+					Version:  "v1",
+					Resource: "namespaces",
+				})
+				ns, err := namespaces.Get(context.TODO(), unstrob.GetNamespace(), metav1.GetOptions{})
+				if err != nil {
+					klog.Error(err)
+					return
+				}
+				if ns == nil {
```

Comment:
We typically don't check that the object is non-nil if err comes back nil. No harm in doing this, just not something you'll usually see in the code.